### PR TITLE
🎉 New Source: PyPI [low-code cdk]

### PR DESCRIPTION
**Pull request by NoelJacob**
## What
Data from PyPI as source

## How
Used Low-code API

## Recommended reading order
1. `spec.yaml`
2. `pypi.yaml`

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Acceptance</strong></summary>

```
Test session starts (platform: linux, Python 3.9.11, pytest 6.2.5, pytest-sugar 0.9.5)
rootdir: /test_input
plugins: hypothesis-6.54.6, cov-3.0.0, mock-3.6.1, sugar-0.9.5, timeout-1.4.2, requests-mock-1.9.3
collecting ... 
 test_core.py ✓✓✓✓✓✓✓✓✓✓s✓✓✓✓✓✓✓✓✓✓✓s✓✓                                                                                                                    96% █████████▋
 test_full_refresh.py ✓                                                                                                                                   100% ██████████

======================================================================== short test summary info ========================================================================
SKIPPED [1] source_acceptance_test/plugin.py:63: Skipping TestIncremental.test_two_sequential_reads: This connector does not implement incremental sync
SKIPPED [1] source_acceptance_test/tests/test_core.py:51: The previous connector image could not be retrieved.
SKIPPED [1] source_acceptance_test/tests/test_core.py:229: The previous connector image could not be retrieved.

Results (66.53s):
      24 passed
       2 skipped

```

</details>

### CODE REVIEW

**Comment by NoelJacob** [comment]
Please do not merge, I have documentation to do.

**Comment by NoelJacob** [comment]
@koconder secrets/config is not present in the repo. Should be why it's failing. I'll try to change the acceptance-test file to point to sample config

**Comment by koconder** [comment]
@NoelJacob this is normal as there is no test config file for this connection yet on airbyte side. Please hang tight as i review.

**Review by marcosmarxm — DISMISSED** [review]
Hello @NoelJacob, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

File: docs/integrations/README.md [hunk 1]
```diff
@@ -187,7 +188,7 @@ For more information about the grading system, see [Product Release Stages](http
 | [Whisky Hunter](sources/whisky-hunter.md        )                                           | Alpha                | No                  |
 | [WooCommerce](sources/woocommerce.md)                                                       | Alpha                | No                  |
 | [Wordpress](sources/wordpress.md)                                                           | Alpha                | No                  |
-| [Workable](sources/workable.md)                                                           | Alpha                | No                  |
+| [Workable](sources/workable.md)                                                             | Alpha                | No                  |
```
Comment (koconder) [inline docs/integrations/README.md:191]: @NoelJacob please revert this change

File: airbyte-integrations/builds.md [hunk 2]
```diff
@@ -2,135 +2,136 @@
...
-| Zendesk Talk   | [![source-zendesk-talk](https://img.shields.io/endpoint?url=https%3A%2F%2Fdnsgjos7lj2fu.cloudfront.net%2Ftests%2Fsummary%2Fsource-zendesk-talk%2Fbadge.json)](https://dnsgjos7lj2fu.cloudfront.net/tests/summary/source-zendesk-talk) |
-| Zoom   | [![source-zoom-singer](https://img.shields.io/endpoint?url=https%3A%2F%2Fdnsgjos7lj2fu.cloudfront.net%2Ftests%2Fsummary%2Fsource-zoom-singer%2Fbadge.json)](https://dnsgjos7lj2fu.cloudfront.net/tests/summary/source-zoom-singer) |
-| Zuora   | [![source-zuora](https://img.shields.io/endpoint?url=https%3A%2F%2Fdnsgjos7lj2fu.cloudfront.net%2Ftests%2Fsummary%2Fsource-zuora%2Fbadge.json)](https://dnsgjos7lj2fu.cloudfront.net/tests/summary/source-zuora) |
+| name                           | status                                                                                                                                                                                                                                                                                      |
```
Comment (koconder) [inline airbyte-integrations/builds.md:5]: @NoelJacob revert all changes on all lines, just keep your changes

**Review by vincentkoc — CHANGES_REQUESTED** [review]
@NoelJacob would be good if you look at some of the minor changes. I will go ahead and do a quick local acceptance test and see where we go from there. Thanks again for the contribution

**Comment by koconder** [comment]
Test results:

```
=================================== FAILURES ===================================
_______________________ TestBasicRead.test_read[inputs0] _______________________

self = <source_acceptance_test.tests.test_core.TestBasicRead object at 0xffffb6af6070>
connector_config = SecretDict(******)
configured_catalog = ConfiguredAirbyteCatalog(streams=[ConfiguredAirbyteStream(stream=AirbyteStream(name='project', json_schema={'$schema':...l_refresh'>, cursor_field=None, destination_sync_mode=<DestinationSyncMode.overwrite: 'overwrite'>, primary_key=None)])
inputs = BasicReadTestConfig(config_path='secrets/config.json', configured_catalog_path='integration_tests/configured_catalog.j...ds=True), validate_schema=True, validate_data_points=False, expect_trace_message_on_failure=True, timeout_seconds=None)
expected_records_by_stream = defaultdict(<class 'list'>, {'stats': [{'top_packages': {'CodeIntel': {'size': 23767329521}, 'MegEngine': {'size': 151...82ecf9e0e43af740c79ccd/sampleproject-2.0.0.tar.gz', 'yanked': False, 'yanked_reason': None}], 'vulnerabilities': []}]})
docker_runner = <source_acceptance_test.utils.connector_runner.ConnectorRunner object at 0xffffb6c35bb0>
detailed_logger = <Logger detailed_logger /test_input/acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt (DEBUG)>

    def test_read(
        self,
        connector_config,
        configured_catalog,
        inputs: BasicReadTestConfig,
        expected_records_by_stream: MutableMapping[str, List[MutableMapping]],
        docker_runner: ConnectorRunner,
        detailed_logger,
    ):
        output = docker_runner.call_read(connector_config, configured_catalog)
        records = [message.record for message in filter_output(output, Type.RECORD)]
    
        assert records, "At least one record should be read using provided catalog"
    
        if inputs.validate_schema:
>           self._validate_schema(records=records, configured_catalog=configured_catalog)

/usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:480: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

records = [AirbyteRecordMessage(namespace=None, stream='project', data={'info': {'author': 'Hipages Data Team', 'author_email': ...ze': 27927907201}, 'xpress': {'size': 20577744315}}, 'total_packages_size': 14300830071491}, emitted_at=1667482761421)]
configured_catalog = ConfiguredAirbyteCatalog(streams=[ConfiguredAirbyteStream(stream=AirbyteStream(name='project', json_schema={'$schema':...l_refresh'>, cursor_field=None, destination_sync_mode=<DestinationSyncMode.overwrite: 'overwrite'>, primary_key=None)])

    @staticmethod
    def _validate_schema(records: List[AirbyteRecordMessage], configured_catalog: ConfiguredAirbyteCatalog):
        """
        Check if data type and structure in records matches the one in json_schema of the stream in catalog
        """
        TestBasicRead._validate_records_structure(records, configured_catalog)
        bar = "-" * 80
        streams_errors = verify_records_schema(records, configured_catalog)
        for stream_name, errors in streams_errors.items():
            errors = map(str, errors.values())
            str_errors = f"\n{bar}\n".join(errors)
            logging.error(f"\nThe {stream_name} stream has the following schema errors:\n{str_errors}")
    
        if streams_errors:
>           pytest.fail(f"Please check your json_schema in selected streams {tuple(streams_errors.keys())}.")
E           Failed: Please check your json_schema in selected streams ('project', 'release').

/usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:382: Failed
----------------------------- Captured stdout call -----------------------------
{"type": "LOG", "log": {"level": "ERROR", "message": "\nThe project stream has the following schema errors:\nNone is not of type 'string'\n\nFailed validating 'type' in schema['properties']['info']['properties']['platform']:\n    {'description': '[DEPRECATED]', 'type': 'string'}\n\nOn instance['info']['platform']:\n    None\n--------------------------------------------------------------------------------\nNone is not of type 'object'\n\nFailed validating 'type' in schema['properties']['info']['properties']['project_urls']:\n    {'description': 'Additional URLs that are relevant to your project. '\n                    'Corresponds to '\n                    'https://packaging.python.org/specifications/core-metadata/#project-url-multiple-use',\n     'patternProperties': {'.*': {'type': 'string'}},\n     'type': 'object'}\n\nOn instance['info']['project_urls']:\n    None"}}
{"type": "LOG", "log": {"level": "ERROR", "message": "\nThe release stream has the following schema errors:\nNone is not of type 'string'\n\nFailed validating 'type' in schema['properties']['info']['properties']['platform']:\n    {'description': '[DEPRECATED]', 'type': 'string'}\n\nOn instance['info']['platform']:\n    None\n--------------------------------------------------------------------------------\nNone is not of type 'object'\n\nFailed validating 'type' in schema['properties']['info']['properties']['project_urls']:\n    {'description': 'Additional URLs that are relevant to your project. '\n                    'Corresponds to '\n                    'https://packaging.python.org/specifications/core-metadata/#project-url-multiple-use',\n     'patternProperties': {'.*': {'type': 'string'}},\n     'type': 'object'}\n\nOn instance['info']['project_urls']:\n    None"}}
------------------------------ Captured log call -------------------------------
ERROR    root:test_core.py:379 
The project stream has the following schema errors:
None is not of type 'string'

Failed validating 'type' in schema['properties']['info']['properties']['platform']:
    {'description': '[DEPRECATED]', 'type': 'string'}

On instance['info']['platform']:
    None
--------------------------------------------------------------------------------
None is not of type 'object'

Failed validating 'type' in schema['properties']['info']['properties']['project_urls']:
    {'description': 'Additional URLs that are relevant to your project. '
                    'Corresponds to '
                    'https://packaging.python.org/specifications/core-metadata/#project-url-multiple-use',
     'patternProperties': {'.*': {'type': 'string'}},
     'type': 'object'}

On instance['info']['project_urls']:
    None
ERROR    root:test_core.py:379 
The release stream has the following schema errors:
None is not of type 'string'

Failed validating 'type' in schema['properties']['info']['properties']['platform']:
    {'description': '[DEPRECATED]', 'type': 'string'}

On instance['info']['platform']:
    None
--------------------------------------------------------------------------------
None is not of type 'object'

Failed validating 'type' in schema['properties']['info']['properties']['project_urls']:
    {'description': 'Additional URLs that are relevant to your project. '
                    'Corresponds to '
                    'https://packaging.python.org/specifications/core-metadata/#project-url-multiple-use',
     'patternProperties': {'.*': {'type': 'string'}},
     'type': 'object'}

On instance['info']['project_urls']:
    None
=========================== short test summary info ============================
FAILED test_core.py::TestBasicRead::test_read[inputs0] - Failed: Please check...
SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/plugin.py:63: Skipping TestIncremental.test_two_sequential_reads: This connector does not implement incremental sync
SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:51: The previous connector image could not be retrieved.
SKIPPED [1] ../usr/local/lib/python3.9/site-packages/source_acceptance_test/tests/test_core.py:229: The previous connector image could not be retrieved.
=================== 1 failed, 23 passed, 3 skipped in 27.14s ===================
```

**Review by vincentkoc — APPROVED** [review]

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**